### PR TITLE
build: fix standalone compile (3/3T Nougat)

### PR DESCRIPTION
**Pull request by NewEraCracker**
This kernel does not build unless the locations of the includes are fixed, and the compiler wrapper disabled.
This PR fixes those problems.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -82,6 +82,11 @@ GRTAGS
 GSYMS
 GTAGS
 
+*.rej
+*.diff
+*.patch
+*.list
+out/*
```
Comment (gregkh) [inline .gitignore:89]: This file shouldn't be changed :(

File: Makefile [hunk 2]
```diff
@@ -371,13 +371,13 @@ CHECK		= sparse
 
 # Use the wrapper for the compiler.  This wrapper scans for new
 # warnings and causes the build to stop upon encountering them.
-CC		= $(srctree)/scripts/gcc-wrapper.py $(REAL_CC)
+# CC		= $(srctree)/scripts/gcc-wrapper.py $(REAL_CC)
```
Comment (gregkh) [inline Makefile]: Why this change?

File: Makefile [hunk 3]
```diff
@@ -371,13 +371,13 @@ CHECK		= sparse
...
 
 CHECKFLAGS     := -D__linux__ -Dlinux -D__STDC__ -Dunix -D__unix__ \
 		  -Wbitwise -Wno-return-void $(CF)
 CFLAGS_MODULE   =
 AFLAGS_MODULE   =
-LDFLAGS_MODULE  =
+LDFLAGS_MODULE  = --strip-debug
```
Comment (gregkh) [inline Makefile]: Why this?

File: drivers/net/ethernet/msm/rndis_ipa.c [hunk 4]
```diff
@@ -29,7 +29,7 @@
 #include <linux/workqueue.h>
 
 #define CREATE_TRACE_POINTS
-#include "rndis_ipa_trace.h"
+#include <trace/rndis_ipa_trace.h>
```
Comment (gregkh) [inline drivers/net/ethernet/msm/rndis_ipa.c:32]: I understand the need/want for this type of change, but the way kernels are built in an android system means this isn't needed.  So I don't think this is really necessary.

But it would be nice :)

Same with all of the other include changes in this patch.

File: drivers/input/touchscreen/synaptics_driver_s3320.c [hunk 5]
```diff
@@ -3952,8 +3952,8 @@ static int synaptics_ts_probe(struct i2c_client *client, const struct i2c_device
 	TP_FW = CURRENT_FIRMWARE_ID;
 	sprintf(ts->fw_id,"0x%x",TP_FW);
 
-	memset(ts->fw_name,TP_FW_NAME_MAX_LEN,0);
-	memset(ts->test_limit_name,TP_FW_NAME_MAX_LEN,0);
+	memset(ts->fw_name,0,TP_FW_NAME_MAX_LEN);
```
Comment (gregkh) [inline drivers/input/touchscreen/synaptics_driver_s3320.c:3955]: This isn't a build-break issue, it's a totally separate one.  Don't lump multiple different things into the same patch, it makes it hard to apply, and the upstream kernel developers will flat out reject it.

**Review by gregkh — CHANGES_REQUESTED** [review]
Um, needs work, see individual comments...

File: drivers/input/touchscreen/synaptics_driver_s3320.c (hunk 5 above)
Comment (NewEraCracker) [reply drivers/input/touchscreen/synaptics_driver_s3320.c:3955]: It breaks build with gcc-linaro-5.3.1-2016.05 because memset transposed args warning.
So I fixed it.

File: Makefile (hunk 2 above)
Comment (NewEraCracker) [reply Makefile]: This and the REAL_CC change disable the nasty python wrapper for GCC that stops build on certain warnings.

File: Makefile (hunk 3 above)
Comment (NewEraCracker) [reply Makefile]: This will reduce the size of the loadable modules.

Comment (gregkh) [reply Makefile]: That's fine, but what if you wanted the debug symbols?  :)

And it has nothing to do with the original goal of this patch, so please strip it out and submit it separately, so I can reject it there :)  These options can be changed on the command line if you really want it, don't go hard-coding platform specific things into the makefile, that's not ok.

File: drivers/input/touchscreen/synaptics_driver_s3320.c (hunk 5 above)
Comment (gregkh) [reply drivers/input/touchscreen/synaptics_driver_s3320.c:3955]: Ok, then that needs to be called out specifically and done as a separate patch please.

**Comment by jcadduono** [comment]
hmm, I'm building standalone and the only change actually needed for compiling with 5.3.1 toolchain was the transposed arguments. other than that, you can add
```
KBUILD_CFLAGS   += $(call cc-disable-warning,maybe-uninitialized,)
KBUILD_CFLAGS   += $(call cc-disable-warning,unused-const-variable,)
```
right below
```
include $(srctree)/arch/$(SRCARCH)/Makefile
```
in Makefile.

It is probably better to disable build failures on those warnings rather than disable the warnings entirely, however. Not sure what the call for disabling specific -Werror features is.

for the kernel modules stripping, you can call `make modules_install` with `INSTALL_MOD_STRIP=1 INSTALL_MOD_PATH="."`
this should strip and install the modules to ./lib/modules

if it makes it easier for you, I have made some scripts to accomplish building with a few less keystrokes here:
https://github.com/jcadduono/android_kernel_oneplus_msm8996/commit/24038d9b962a37550f76c522f683e2aee54bc501
(add .patch to the url for a patch)

@gregkh i would not advise visiting my repositories, you would be critiquing for days, bad commit habits i'm guilty :)
cool to see you here, i'd say i'm somewhat of a fan!

i'm not a big fan of the android build system and prefer to compile kernels standalone, especially in the way I use them (ex. not built with a device operating system, but simply as a quick kernel image swap: https://github.com/jcadduono/lazyflasher)
